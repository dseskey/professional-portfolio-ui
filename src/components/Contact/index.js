import React, { useState } from 'react';
import { Container, Row, Col, Form, Button, Alert } from 'react-bootstrap';
import { validateEmail } from '../../utils/helpers';
function Contact() {
    const [formState, setFormState] = useState({ name: '', email: '', message: '' });
    const [errorMessage, setErrorMessage] = useState('');

    function handleChange(event) {
        if (event.target.name === 'email') {
            const isValid = validateEmail(event.target.value);
            if (!isValid) {
                setErrorMessage('The email that you have provided is invalid.');
            } else {
                setErrorMessage('');
            }
        } else {
            if (!event.target.value.length) {
                setErrorMessage(`A ${event.target.name} is required.`);
            } else {
                setErrorMessage('');
            }
        }
        if (!errorMessage) {
            setFormState({ ...formState, [event.target.name]: event.target.value });
        }
    
    }

    function handleSubmit(event) {
        event.preventDefault();
        console.log(formState);
    }

    return (
        <Container fluid data-testid='contact-container' id='contact-container' className='pagesParentContainer'>
            <Row className='pageHeader'>
                <h2 data-testid='contact-form-header'>Contact me</h2>
            </Row>
            <Row id='contact-form-row'>
                <Form>
                    <Form.Row>
                        <Form.Group as={Col} controlId='formGirdName'>
                            <Form.Label>Name</Form.Label>
                            <Form.Control placeholder='Enter Name' name='name' onBlur={(event) => handleChange(event)} />
                        </Form.Group>
                        <Form.Group as={Col} controlId='formGridEmail'>
                            <Form.Label>Email</Form.Label>
                            <Form.Control type='email' placeholder='Enter email' name='email' onBlur={(event) => handleChange(event)} />
                        </Form.Group>
                    </Form.Row>
                    <Form.Row>
                        <Form.Label>Message</Form.Label>
                        <Form.Control data-testid='message-text-area' as='textarea' placeholder='message' name='message' onBlur={(event) => handleChange(event)} />
                    </Form.Row>
                    <Form.Row>
                        {errorMessage && (
                            <Alert variant='danger' className='errorContainer'>
                                {errorMessage}
                            </Alert>
                        )}
                    </Form.Row>
                    <Button data-testid='submit-button' variant='primary' type='submit' id='submit-contact-button' onClick={(event) => handleSubmit(event)}>
                        Submit
                    </Button>
                </Form>
            </Row>
        </Container>
    )
}

export default Contact;