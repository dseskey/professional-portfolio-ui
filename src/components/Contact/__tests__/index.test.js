import React from 'react';
import {render, cleanup, fireEvent} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Contact from '..';


afterEach(cleanup);

describe('Contact component', () =>{
    it('renders', () =>{
        render(<Contact/>);
    })
    it('matches snapshot', () => {
        const {asFragment} = render(<Contact />);
        expect(asFragment()).toMatchSnapshot();
    })
})

describe('Contact header is visible', () => {
    it('shows Contact header', () => {
        const {getByTestId} = render(<Contact/>);
        expect(getByTestId('contact-form-header')).toHaveTextContent('Contact Me');
    });
});

describe('Submit button is visible', () =>{
    it('contains Submit label', () =>{ 
        const {getByTestId} = render(<Contact/>);
        expect(getByTestId('submit-button')).toHaveTextContent('Submit');
    })
})