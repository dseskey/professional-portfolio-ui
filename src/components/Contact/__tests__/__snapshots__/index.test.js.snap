// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="pagesParentContainer container-fluid"
    data-testid="contact-container"
    id="contact-container"
  >
    <div
      class="pageHeader row"
    >
      <h2
        data-testid="contact-form-header"
      >
        Contact me
      </h2>
    </div>
    <div
      class="row"
      id="contact-form-row"
    >
      <form
        class=""
      >
        <div
          class="form-row"
        >
          <div
            class="form-group col"
          >
            <label
              class="form-label"
              for="formGirdName"
            >
              Name
            </label>
            <input
              class="form-control"
              id="formGirdName"
              name="name"
              placeholder="Enter Name"
            />
          </div>
          <div
            class="form-group col"
          >
            <label
              class="form-label"
              for="formGridEmail"
            >
              Email
            </label>
            <input
              class="form-control"
              id="formGridEmail"
              name="email"
              placeholder="Enter email"
              type="email"
            />
          </div>
        </div>
        <div
          class="form-row"
        >
          <label
            class="form-label"
          >
            Message
          </label>
          <textarea
            class="form-control"
            data-testid="message-text-area"
            name="message"
            placeholder="message"
          />
        </div>
        <div
          class="form-row"
          data-testid="contact-error-container"
        />
        <button
          class="btn btn-primary"
          data-testid="submit-button"
          id="submit-contact-button"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
